### PR TITLE
_WIN32_WCE port of wolfCrypt - OE12

### PR DESCRIPTION
**Pull request by kaleb-himes**

### CODE REVIEW

**Comment by toddouska** [comment]
@dgarske please review.  After approval assign me.

File: wolfcrypt/benchmark/benchmark.c [hunk 1]
```diff
@@ -4573,18 +4573,21 @@ static void bench_rsa_helper(int doAsync, RsaKey rsaKey[BENCH_MAX_PENDING],
...
         byte* enc[BENCH_MAX_PENDING];
     #endif
     #if !defined(WOLFSSL_RSA_VERIFY_INLINE) && !defined(WOLFSSL_RSA_PUBLIC_ONLY)
-        #if !defined(WOLFSSL_MDK5_COMPLv5)
+        #if !defined(WOLFSSL_MDK5_COMPLv5) && !defined(_WIN32_WCE)
           /* MDK5 compiler regard this as a executable statement, and does not allow declarations after the line. */
             DECLARE_ARRAY_DYNAMIC_DEC(out, byte, BENCH_MAX_PENDING, rsaKeySz, HEAP_HINT);
             #else
-                int idxout;
+              int idxout;
```
Comment (dgarske) [inline wolfcrypt/benchmark/benchmark.c:4589]: I don't see `int inner_idx_out;` and `int idxout;`  used anywhere.... If it is required might make sense to have a single block for them.

File: wolfcrypt/src/wc_port.c [hunk 2]
```diff
@@ -441,7 +441,11 @@ int wc_ReadDirFirst(ReadDirCtx* ctx, const char* path, char** name)
     XSTRNCPY(ctx->name, path, MAX_FILENAME_SZ - 3);
     XSTRNCPY(ctx->name + pathLen, "\\*", MAX_FILENAME_SZ - pathLen);
 
+    #ifdef _WIN32_WCE
+    ctx->hFind = FindFirstFileA((LPCWSTR) ctx->name, &ctx->FindFileData);
```
Comment (dgarske) [inline wolfcrypt/src/wc_port.c:445]: wc_port.h has a macro for `FindFirstFileA ` and `_WIN32_WCE`. Seems like these differences should be solved in the macro...?

`#define FindFirstFileA(fn, d) FindFirstFile(fn, (LPWIN32_FIND_DATAW) d)`

File: wolfssl/wolfcrypt/types.h [hunk 3]
```diff
@@ -597,7 +597,11 @@ decouple library dependencies with standard string, memory and so on.
 
                         if ((int)bufsize <= 0) return -1;
                         va_start(ap, format);
+                    #ifndef _WIN32_WCE
```
Comment (dgarske) [inline wolfssl/wolfcrypt/types.h:600]: Can this be fixed to use our `XVSNPRINTF` macro and changed in wc_port.h?

**Review by dgarske — CHANGES_REQUESTED** [review]

File: wolfcrypt/benchmark/benchmark.c (hunk 1 above)
Comment (kaleb-himes) [reply wolfcrypt/benchmark/benchmark.c:4589]: Good note, I didn't add those variables but yes I can see they are not used anywhere, happy to clean them up.

File: wolfcrypt/src/wc_port.c (hunk 2 above)
Comment (kaleb-himes) [reply wolfcrypt/src/wc_port.c:445]: Fantastic suggestion. Done!

File: wolfssl/wolfcrypt/types.h (hunk 3 above)
Comment (kaleb-himes) [reply wolfssl/wolfcrypt/types.h:600]: Absolutely!

**Review by dgarske — APPROVED** [review]
Thanks. Over to Todd for final.

**Comment by kaleb-himes** [comment]
retest this please.

File: wolfssl/wolfcrypt/wc_port.h [hunk 4]
```diff
@@ -863,9 +879,36 @@ WOLFSSL_API int wolfCrypt_Cleanup(void);
...
     #include <windows.h>
+    #include <stdlib.h> /* For file system */
+
+    time_t windows_time(time_t* timer);
+
+    #define FindNextFileA(h, d) FindNextFile(h, (LPWIN32_FIND_DATAW) d)
+    #ifdef _WIN32_WCE
+        #define FindFirstFileA(fn, d) FindFirstFile((LPCWSTR) fn, \
+                                                    (LPWIN32_FIND_DATAW) d)
+    #else
```
[the remark this answers is not in the document]
Comment (kaleb-himes) [reply wolfssl/wolfcrypt/wc_port.h]: Cleaned up, thank you @dgarske .

**Review by toddouska — APPROVED** [review]

**Pull request merged**